### PR TITLE
Test for edge cases

### PR DESCRIPTION
**Pull request by rianp**
**summary or list of acceptance criteria**
1. make sure invalid input doesn’t crash the game
2. Add tests for edge cases, like updating a cell in the board that doesn’t exist and ZOMBIES tests.

**summary or list of what you updated & added**
1. add tests for input validation
2. break up input validation method into 3 separate methods
3. change error handling messages for input validation
4. add get board range method and test for board class

[ticket link](https://trello.com/c/Jb2hFuTv)

### CODE REVIEW

File: game.py [hunk 1]
```diff
@@ -55,20 +55,26 @@ def display_instructions():
...
         validator = Validator()
-        while not validator.validate_selection(user_input, self._board.get_board()):
-            Console.print_string("Oops! This is an invalid menu choice. ")
-            user_input = Console.prompt_input("\nLet's try that again! Enter a value please: ")
-        return user_input
+        while True:
+            if not validator.is_valid_integer(user_input):
+                Console().print_string("Eek! That's not even a number! ")
+            elif not validator.is_in_range(int(user_input), self._board.get_board_range()):
+                Console().print_string("Whoa friend! This is outa bounds! ")
```
Comment (fifikim) [inline game.py]: non-code related nit! "outta" not "outa"

love these messages though

File: test_validation.py [hunk 2]
```diff
@@ -8,15 +8,43 @@ def setUp(self):
...
+        with self.subTest('returns false when input is a character'):
+            result = self.validator.is_valid_integer("r")
+            self.assertFalse(result)
+
+        with self.subTest('returns false when input is none'):
+            result = self.validator.is_valid_integer("")
+            self.assertFalse(result)
+
+        with self.subTest('returns false when input is a space'):
+            result = self.validator.is_valid_integer(" ")
```
Comment (fifikim) [inline test_validation.py:49]: how about when there's whitespace before and/or after a valid integer?

File: test_validation.py [hunk 3]
```diff
@@ -8,15 +8,43 @@ def setUp(self):
...
-        with self.subTest('returns false when out of range integers'):
-            result = self.validator.validate_selection(10, self.test_board)
+        with self.subTest('returns false when selection is not available'):
+            result = self.validator.is_on_board(4, test_board)
             self.assertFalse(result)
 
-        with self.subTest('returns false when non integers'):
-            result = self.validator.validate_selection("r", self.test_board)
+    def test_is_in_range(self):
+        board_range = range(1, len(self.test_board) * len(self.test_board) + 1)
```
Comment (fifikim) [inline test_validation.py:22]: really nice job using logical describe blocks, consistent patterns in your names, and descriptive test cases/outcomes 👏👏👏

File: game.py [hunk 4]
```diff
@@ -15,24 +15,24 @@ def __init__(self):
         self._turn = Turn()
 
     def play(self):
-        Console.print_string("Hello friend, welcome to Tic-Tac-Toe!")
+        Console().print_string("Hello friend, welcome to Tic-Tac-Toe!")
         self.display_instructions()
```
Comment (ranizilpelwar) [inline game.py:19]: Instead of initializing Console every time, would it be valuable to have an instance in your class that you can refer to? Similar to "self._console = Console()"?

File: game.py (hunk 1 above)
Comment (ranizilpelwar) [reply game.py]: +1 love these messages

File: test_validation.py (hunk 3 above)
Comment (ranizilpelwar) [reply test_validation.py:22]: +1000

Comment (ranizilpelwar) [reply test_validation.py:22]: I also really love that you took the feedback for PRs and ran with it, consistently with every PR you have done since. The Descriptions are great.

File: game.py (hunk 4 above)
Comment (ranizilpelwar) [reply game.py:19]: Note: This is not a PR merge blocker, just an idea/suggestion.

Comment (rianp) [reply game.py:19]: definitely! I was planning on getting to this with the game tests ticket and the inevitable refactor of game class that will come with those tests.

Comment (ranizilpelwar) [reply game.py:19]: awesome!

**Review by ranizilpelwar — APPROVED** [review]
